pkg/sqlgen/sqlgen.go

```diff
@@ -104,18 +104,41 @@ func BuildMetaTableFromInput(
 }
 
 var base64EncodedFieldTypes = []schema.FieldType{schema.FTByteString, schema.FTBinary}
+var stringEncodedFieldTypes = []schema.FieldType{schema.FTString, schema.FTText}
+
+func isStringEncodedFieldType(fieldType schema.FieldType) bool {
+	return isFieldTypeInSet(fieldType, stringEncodedFieldTypes)
+}
 
 func isBase64EncodedFieldType(fieldType schema.FieldType) bool {
-	for _, eft := range base64EncodedFieldTypes {
-		if fieldType == eft {
+	return isFieldTypeInSet(fieldType, base64EncodedFieldTypes)
+}
+
+func isFieldTypeInSet(fieldType schema.FieldType, fieldTypes []schema.FieldType) bool {
+	for _, thisFieldType := range fieldTypes {
+		if fieldType == thisFieldType {
 			return true
 		}
 	}
 	return false
 }
 
-func maybeDecodeBase64(val interface{}, should bool) (interface{}, error) {
-	if !should || val == nil {
+func convertToByteSliceIfStringHasNulls(val interface{}) (interface{}, error) {
+	if val == nil {
```

Comment:
will this get tripped up by the interface{} nils thing? https://golang.org/doc/faq#nil_error

Reply:
(probably need to consider this elsewhere in this path JIC)

Reply:
I don't think so, as this was logic that was lifted by the previous maybeConvertToBase64 method. i'll take a look tho.

Reply:
Very good chance I didn't understand the scope of the risk there when I wrote that function :)

Reply:
I did a review of where those values get set as interface{} up the stack and they are populated ultimately by the encoding/json package at the executive API, so it should be safe. I talked to Rick and we agreed that we'll leave this behavior as is for this PR. In the future we may add additional checks here.